### PR TITLE
ETL: Push data to socrata

### LINKED ISSUE #199: VZD: Push crash data to Socrata
- Crash table
- Persons table - this will allow us to calculate years of life lost

Potential use Python client helpers.
- socratautil
- https://github.com/xmunoz/sodapy

> before you can upsert data to a Socrata dataset, you need to specify a “row ID” in the dataset metdata. Docs: https://dev.socrata.com/docs/row-identifiers.html

### LINKED ISSUE #357: VZV: Publish data for Serious Injury + Fatal Crashes by Mode chart
The data for this chart is currently hard-coded with figures from the paper Vision Zero Dashboard released by the Vision Zero team (data up to 8/26/(18 & 19)).

Data about modes involved in crashes needs to be published to Socrata to populate this chart with data programatically.

### Chart
Screen Shot 2019-10-28 at 12 16 40 PM

### PR DESCRIPTION
**Pull request by mddilley**
Closes #199
Closes #357

### CODE REVIEW

**Comment by mddilley** [comment]
This PR add a process to request records from Hasura, transform the data, and then publish to a Socrata dataset. A list of configs in the script defines which table and query to use, configs for the data transformation, and the dataset 4x4 to publish records. I'd appreciate any feedback on error handling, logging, and anything else that stands out.

Thanks to @sergio for the help getting started and also refactoring the commands to execute the ETL scripts. That made it much easier dive into this issue.

File: atd-etl/app/process/helpers_socrata.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
+"""
+Helpers for Socrata Export
+Author: Austin Transportation Department, Data and Technology Office
```
Comment (mateoclarke) [inline atd-etl/app/process/helpers_socrata.py]: `Data and Technology Services`

File: atd-etl/app/process_socrata_export.py [hunk 2]
```diff
@@ -6,11 +6,99 @@
...
+        "table": "crash",
+        "template": crashes_query_template,
+        "formatter": format_crash_data,
+        "formatter_config": {
+            "tables": ["atd_txdot_crashes"],
+            "columns_to_rename": {
+                "veh_body_styl_desc": "unit_desc",
+                "veh_unit_desc_desc": "unit_mode",
+            },
+            "flags_list": ["MOTOR VEHICLE",
```
Comment (mateoclarke) [inline atd-etl/app/process_socrata_export.py]: I like the way this is implemented. that is all.

File: atd-etl/app/process/socrata_queries.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+Important: These templates require importing the Template class from the string library.
+"""
+from string import Template
+
+# Queries Hasura to retrieve records for
+# https://data.austintexas.gov/d/y2wy-tgr5
+crashes_query_template = Template(
+    """
+    query getCrashesSocrata {
+        atd_txdot_crashes (limit: $limit, offset: $offset, order_by: {crash_id: asc}, where: {city_id: {_eq: 22}}) {
```
Comment (mateoclarke) [inline atd-etl/app/process/socrata_queries.py]: just putting a bookmark here so you remember that if we change how we query by city id we'll need to modify this code.

File: atd-etl/app/process/helpers_socrata.py (hunk 1 above)
Comment (mddilley) [reply atd-etl/app/process/helpers_socrata.py]: Fixed! 🙃

File: atd-etl/app/process_socrata_export.py (hunk 2 above)
Comment (mddilley) [reply atd-etl/app/process_socrata_export.py]: Thanks! I was pretty happy that I was able refactor this way. I was hoping to come up with one formatter for any table type but that could come down the road with other datasets  that we publish.

File: atd-etl/app/process/socrata_queries.py (hunk 3 above)
Comment (mddilley) [reply atd-etl/app/process/socrata_queries.py]: Awesome, I'll keep this in mind too if we decide to add more `city_id`s.

**Pull request merged**